### PR TITLE
[4.0] Not only set the redirect but actually return

### PR DESCRIPTION
**Pull request by Hackwar**
setRedirect() only sets where to redirect to, it doesn't actually redirect. So we have to return from the method to actually abort operations here.

### CODE REVIEW

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 1]
```diff
@@ -70,23 +70,21 @@ public function cancel()
 	public function save()
 	{
 		// Check for request forgeries.
-		if (!Session::checkToken())
-		{
-			$this->setRedirect('index.php', Text::_('JINVALID_TOKEN'), 'error');
-		}
+		Session::checkToken() or jexit(Text::_('JINVALID_TOKEN'));
```
Comment (wilsonge) [inline administrator/components/com_config/Controller/ApplicationController.php]: Please just use `$this->checkToken()` https://github.com/joomla/joomla-cms/blob/2ab3309cae09c3cd64025ffb1e26e2728e2b9868/libraries/src/MVC/Controller/BaseController.php#L1076

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 2]
```diff
@@ -70,23 +70,21 @@ public function cancel()
...
+			return false;
 		}
 
 		// Set FTP credentials, if given.
 		ClientHelper::setCredentialsFromRequest('ftp');
 
 		/** @var \Joomla\Component\Config\Administrator\Model\ApplicationModel $model */
 		$model = $this->getModel('Application', 'Administrator');
-
+		$app   = Factory::getApplication();
```
Comment (wilsonge) [inline administrator/components/com_config/Controller/ApplicationController.php]: `$this->app`

File: administrator/components/com_config/Controller/ApplicationController.php [hunk 3]
```diff
@@ -135,6 +146,7 @@ public function save()
 
 			// Save failed, go back to the screen and display a notice.
 			$this->app->redirect(Route::_('index.php?option=com_config', false));
+			return false;
```
Comment (wilsonge) [inline administrator/components/com_config/Controller/ApplicationController.php:149]: `JApplicationWeb::redirect()` actually does a redirect. So either we don't need to return false in theory. Or we should setRedirect and return false. I'm not sure why we setRedirect in some places and redirect in another places :/

Comment (Hackwar) [reply administrator/components/com_config/Controller/ApplicationController.php:149]: You are right, $this->app->redirect() does redirect. The idea afaik is, that you can catch a redirect between setting it and actually redirecting in the components entry file.

Comment (wilsonge) [reply administrator/components/com_config/Controller/ApplicationController.php:149]: Then let's use `setRedirect` here and not actually redirect

**Comment by Hackwar** [comment]
All requested changes should be applied.

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
Thanks :)

**Pull request merged**